### PR TITLE
chore(windows): add unit test for SetupCoreEnvironment

### PR DESCRIPTION
**Pull request by rc-swag**
This is just a simple test for SetupCoreEnvironment.  Now actually covered by the keyman developer test.

### CODE REVIEW

**Comment by rc-swag** [comment]
@keymanapp-test-bot skip

File: core/src/mock/mock_processor.cpp [hunk 1]
```diff
@@ -96,11 +96,21 @@ namespace km {
...
+      if (i == _options.end()){
+        auto j = _options.emplace(char16_t(scope) + key,value);
+        if(j.second){
+          persisted_store()[key] = value;
+          return option(scope, key, j.first->second);
+        }
+        else {
+          return option();
+       }
+      }
```
Comment (mcdurdin) [inline core/src/mock/mock_processor.cpp:108]:
```suggestion
      if(i == _options.end()) {
        auto j = _options.emplace(char16_t(scope) + key, value);
        if(j.second) {
          persisted_store()[key] = value;
          return option(scope, key, j.first->second);
        }
        else {
          return option();
        }
      }
```

Can you also add a comment explaining what this is doing? The test for `j.second` is particularly obscure to me :grin:

**Review by mcdurdin — APPROVED** [review]

File: core/src/mock/mock_processor.cpp [hunk 2]
```diff
@@ -96,11 +96,21 @@ namespace km {
...
-      if (i == _options.end()) return option();
-
-      i->second = value;
-      persisted_store()[key] = value;
-      return option(scope, key, i->second);
+      if(i == _options.end()) {
+        auto j = _options.emplace(char16_t(scope) + key, value);
+        if(j.second) {
+          persisted_store()[key] = value;
+          return option(scope, key, j.first->second);
```
[the remark this answers is not in the document]
Comment (rc-swag) [reply core/src/mock/mock_processor.cpp]: I have rearranged the function changed the variable name and added a small comment hopefully that makes it easier to follow.   The return value of `emplace` is quoting "a pair consisting of an iterator to the inserted element, or the already-existing element if no insertion happened, and a bool denoting whether the insertion took place (true if insertion happened, false if it did not)."

**Review by ermshiperete — APPROVED** [review]

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 16.0.105-beta](https://keyman.com/downloads/releases/beta/16.0.105)

**Pull request merged**